bfvmm/include/hve/arch/intel_x64/vcpu.h

```diff
@@ -126,16 +130,39 @@ class EXPORT_HVE vcpu : public bfvmm::vcpu
     ///
     VIRTUAL void run_delegate(bfobject *obj);
 
-    /// Halt Delegate
+    /// Add Launch Delegate
+    ///
+    /// Adds a launch delegate to the VCPU. The delegates are added to a queue and
+    /// executed in FIFO order. All delegates are executed unless an exception
```

Comment:
Is there a reason this if FIFO and not LIFO like the other delegate lists?

Reply:
The same reason the run, hlt, init and fini delegates don't. In this case, there is some core functionality that has to run, so the delegates are not overridding like the VMexit handlers.